### PR TITLE
Correct indirect P/Invoke call prlogs on x86.

### PR DESCRIPTION
**Pull request by pgavlin**
32-bit targets require the size of the stack argument area in
InlinedCallFrame.m_Datum rather than the stub argument (as on 64-bit
targets).

### CODE REVIEW

**Comment by pgavlin** [comment]
Fixes VSO 279868.

**Comment by pgavlin** [comment]
@BruceForstall @adiaaida @dotnet/jit-contrib PTAL

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -2784,11 +2784,18 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
...
+        const unsigned numStkArgBytes = call->fgArgInfo->GetNextSlotNum() * TARGET_POINTER_SIZE;
+        src = comp->gtNewIconNode(numStkArgBytes, TYP_INT);
+#else
+        // On 64-bit targets, indirect calls may need the stub parameter value in InlinedCallFrame.m_Datum.
         if (comp->info.compPublishStubParam)
         {
-            src = new (comp, GT_LCL_VAR) GenTreeLclVar(TYP_I_IMPL, comp->lvaStubArgumentVar, BAD_IL_OFFSET);
+            src = comp->gtNewLclvNode(comp->lvaStubArgumentVar, TYP_I_IMPL);
         }
         // else { If we don't have secret parameter, m_Datum will be initialized by VM code }
```
Comment (adiaaida) [inline src/jit/lower.cpp]: Can you do something with this comment so its alignment doesn't get changed? Move it above the if perhaps?

**Comment by BruceForstall** [comment]
LGTM

**Comment by adiaaida** [comment]
LGTM.

**Pull request merged**